src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts

```diff
@@ -28,33 +33,67 @@ export interface ContainerResourceMetricSource {
   name: string;
   targetAverageUtilization?: number;
   targetAverageValue?: string;
+  target?: {
+    averageUtilization?: number;
+    averageValue?: string;
+    type?: string;
+  };
 }
 
 export interface ExternalMetricSource {
-  metricName: string;
+  metricName?: string;
   metricSelector?: LabelSelector;
   targetAverageValue?: string;
   targetValue?: string;
+  metric?: {
+    name?: string;
+    selector?: LabelSelector;
+  };
+  target?: {
+    type: string;
+    value?: string;
+    averageValue?: string;
+  };
 }
```

Comment:
Another option could be to have:

```ts
export interface V1ExternalMetricSource { ... }
export interface V2Beta1ExternalMetricSource { ... }
export interface V2Beta2ExternalMetricSource { ... }
export interface V2ExternalMetricSource { ... }

export type ExternalMetricSource = 
  | V1ExternalMetricSource
  | V2Beta1ExternalMetricSource
  | V2Beta2ExternalMetricSource
  | V2ExternalMetricSource;
```

WDYT?

Reply:
Made some refactoring and using separated types for now like you suggested.